src/main/java/com/mercuryCyclists/Sales/service/InStoreSaleService.java

```diff
@@ -41,6 +43,32 @@ public List<InStoreSale> getAllInStoreSales() {
         return inStoreSaleRepository.findAll();
     }
 
+    /**
+     * Get sale by id
+     */
+    public InStoreSale getInstoreSale(Long id) {
+
+        Optional<InStoreSale> s = inStoreSaleRepository.findById(id);
+        if(!s.isPresent()) {
+            throw new IllegalArgumentException("Sale with ID does not exist");
+        }
+        return s.get();
+    }
+
+    /**
+     * Get Product by sale
+     */
+    public ResponseEntity<String> getProductBySaleId(InStoreSale s) {
+        if(s == null) {
+            return new ResponseEntity<>("Invalid Sale Id", HttpStatus.FAILED_DEPENDENCY);
+        }
+        //query product endpoint with productID
+        Map<String, Long> param = new HashMap<>();
+        param.put("productId", s.getProductId());
+        String result = restTemplate.getForObject(GETPRODUCTAPI, String.class, param);
+        return new ResponseEntity<>(result, HttpStatus.OK);
+    }
```

Comment:
This can be shared and chucked in sale service. 
However ive got it in my PR, not sure if you wanna wait or?? Im gonna leave it up to you blokes on solving conflicts :)

Reply:
Going to leave in